src/game.js

```diff
@@ -0,0 +1,97 @@
+var Frame = require('../src/frame.js');
+var TenthFrame = require('../src/tenthFrame.js');
+var Updater = require('../src/updater.js');
+
+function Game() {
+  this.frames = []
+  this.updater = new Updater();
+  this.increment = 0;
+  this.createFrames();
+}
+
+Game.prototype.addFrame = function(frame){
+  this.frames.push(frame);
+}
+
+Game.prototype.createFrames = function(){
+  for (var i = 0; i < 9; i++) {
+     this.addFrame(new Frame());
+  }
+  this.addFrame(new TenthFrame());
+}
+
+Game.prototype.getFrame = function(frameIndex) {
+  return this.frames[frameIndex];
+}
+
+Game.prototype.getTotal = function() {
+  this.update();
+  var total = 0;
+  this.frames.forEach(function(frame){
+    total += frame.getFinalFrameScore();
+  });
+  return total;
+}
+
+Game.prototype.addFrame = function(frame){
+  this.frames.push(frame);
+}
+
+Game.prototype.update = function() {
+  this.updater.update(this.frames);
+}
+
+Game.prototype.inputScore = function(score){
+   var currentFrame = this.getFrame(this.increment);
+   if (this.increment == 9) {
+       if (currentFrame.getFinalFrameScore == null) {
+          return "Game has finished!";
+       }
+
+       if (currentFrame.getFirstScore() == null) {
+           currentFrame.setFirstScore(score);
+           this.update();
+           return;
+       }
+
+       if (currentFrame.getSecondScore() == null) {
+           currentFrame.setSecondScore(score);
+           this.update();
+           return;
+       }
+
+       if (currentFrame.isStrike() || currentFrame.isSpare()) {
+           currentFrame.setBonusScore(score);
+           this.update();
+           return;
+       }
+
+   }
+
+   if (currentFrame.getFirstScore() == null) {
+       currentFrame.setFirstScore(score);
+       this.update();
+       return;
+   }
+
+   if (currentFrame.isStrike()) {
+       this.increment += 1;
+       currentFrame = this.getFrame(this.increment);
+       currentFrame.setFirstScore(score);
+       this.update();
+       return;
+   }
+
+   if (!currentFrame.isStrike()) {
```

Comment:
Is this test necessary? What is the default case?

Reply:
Probably not - I'll remove it and rerun my tests.